scripts/test/SANS/gui_logic/add_runs_presenter_test.py

```diff
@@ -431,4 +432,38 @@ def test_that_make_filename_returns_fullname_if_one_is_given(self):
         self.assertEqual(actual_name, expected_name)
 
 
-if __name__ == '__main__': unittest.main()
+class AddRunsDefaultSettingsTest(unittest.TestCase):
+    def setUp(self):
+        self.presenter = AddRunsPagePresenter(mock.Mock(),
+                                              mock.Mock(),
+                                              mock.Mock(),
+                                              mock.Mock(),
+                                              mock.Mock())
+
+    def test_that_presenter_calls_properties_handler_to_update_directory_on_directory_changed(self):
+        new_dir_name = os.path.join("some", "dir", "path")
+        self.presenter._view.display_save_directory_box = mock.Mock(return_value=new_dir_name)
+        self.presenter.gui_properties_handler.update_default = mock.Mock()
+        self.presenter.set_output_directory = mock.Mock()
+
+        self.presenter._handle_output_directory_changed()
+        self.assertEqual(self.presenter.gui_properties_handler.update_default.call_count, 1,
```

Comment:
You could use mock's `assert_called_once_with` here, which also checks the arguments the function is called with. E.g. 

```python
self.presenter.gui_properties_handler.update_default.assert_called_once_with('add_runs_output_directory', new_dir_name)

Reply:
That's a far better solution